src/Microsoft.Azure.WebJobs.Extensions.Kafka/Diagnostics/SingleEventActivityProvider.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka.Diagnostics
+{
+    internal class SingleEventActivityProvider : ActivityProvider, IDisposable
+    {
+        private readonly IKafkaEventData kafkaEvent;
+
+        private static string SingleKafkaTriggerActivityName { get; } = "SingleKafkaTrigger.Process";
+        public SingleEventActivityProvider(IKafkaEventData kafkaEvent, string consumerGroup) : base(kafkaEvent.Topic, consumerGroup)
+        {
+            this.kafkaEvent = kafkaEvent;
+            this.CreateAndStartActivity();
+        }
+
+        public void CreateAndStartActivity()
+        {
+            if (ActivitySource.HasListeners())
+            {
+                KafkaEventInstrumentation.TryExtractTraceParentId(kafkaEvent, out string traceparent);
+                this.CreateActivity(SingleKafkaTriggerActivityName, ActivityKind.Consumer, traceparent);
```

Comment:
I guess the case when traceparent is empty string or null is missed

Reply:
The create activity function creates a new traceparent in this case.